File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        res = 0
        seen = set()

        for n in nums:
            seen.add(n)

        for n in nums:

            if n-1 not in seen:
                length = 0
                while n + length in seen:
                    length += 1
                res = max(res, length)
            
        return res


            

File: Data Structures & Algorithms/duplicate-integer/submission-3.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {boolean}
     */
    hasDuplicate(nums: number[]): boolean {
        const seen = new Set();

        for(let n of nums) {
            if(seen.has(n)) {
                return true
            }
            seen.add(n)
        }
        return false;
    }
}
